processing/src/main/java/org/apache/druid/math/expr/FunctionalExpr.java

```diff
@@ -190,11 +191,22 @@ public ExprEval eval(ObjectBinding bindings)
     try {
       return function.apply(args, bindings);
     }
-    catch (DruidException | ExpressionValidationException e) {
+    catch (ExpressionValidationException e) {
+      // ExpressionValidationException already contain function name
+      throw DruidException.forPersona(DruidException.Persona.USER)
+                          .ofCategory(DruidException.Category.INVALID_INPUT)
+                          .build(e.getMessage());
+    }
+    catch (Types.InvalidCastException | Types.InvalidCastBooleanException e) {
```

Comment:
why this should be so specific? I think `IAE` should be allowed with a plain rethrow

Reply:
see other comment about translating into invalid input/user exception so it makes the appropriate type of error response and not a system error